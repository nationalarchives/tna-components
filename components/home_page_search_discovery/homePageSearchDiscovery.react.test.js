import React from 'react';
import TestRenderer from 'react-test-renderer';
import HomePageSearchDiscovery from './homePageSearchDiscovery.react';
import Headline from './child_components/headline.react';

let testRenderer, testInstance, testRendererJSON;

beforeAll(() => {
  testRenderer = TestRenderer.create(<HomePageSearchDiscovery />);
  testInstance = testRenderer.root;
  testRendererJSON = testRenderer.toJSON();
});

describe('HomePageSearchDiscovery component', () => {
  it('=> Prop H1 got the right content', () => {
    expect(testInstance.findByType(Headline).props.mainhead).toBe(
      'Search The National Archives and over 2,500 archives across the UK'
    );
  });
});
