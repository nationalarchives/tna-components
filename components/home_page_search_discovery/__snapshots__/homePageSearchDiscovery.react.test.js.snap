// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePageSearchDiscovery component DOM snapshot test 1`] = `
<div
  id="discovery-homepage-search-react-component"
>
  <form
    action="https://discovery.nationalarchives.gov.uk/results/r"
    id="discoveryHomePageSearch"
    method="get"
    name="discoveryHomePageSearch"
    onSubmit={[Function]}
  >
    <div
      className="grid-container"
      role="search"
    >
      <div
        className="Headline"
      >
        <h1>
          Search The National Archives and over 2,500 archives across the UK
        </h1>
        <div
          class={null}
          role="alert"
        />
      </div>
      <div
        className="MainSearch"
      >
        <label>
          Show
        </label>
        <input
          className={null}
          id="search-all-collections"
          name="_q"
          onChange={[Function]}
          onKeyUp={[Function]}
          placeholder="everything that contains these terms"
          type="text"
        />
      </div>
      <div
        className="InputBetween"
      >
        <label>
          between
        </label>
        <input
          className={null}
          id="start-date"
          name="_sd"
          onChange={[Function]}
          onKeyUp={[Function]}
          placeholder="YYYY"
          type="number"
        />
      </div>
      <div
        className="InputAnd"
      >
        <label>
          and
        </label>
        <input
          className={null}
          id="end-date"
          name="_ed"
          onChange={[Function]}
          onKeyUp={[Function]}
          placeholder="YYYY"
          type="number"
        />
      </div>
      <div
        className="InputAcross"
      >
        <label>
          across
        </label>
        <select
          id="held-by"
          name="_hb"
        >
          <option
            value=""
          >
            All records
          </option>
          <option
            value="tna"
          >
            The National Archives only
          </option>
          <option
            value="oth"
          >
            Other archives only
          </option>
        </select>
      </div>
      <div
        className="Search"
      >
        <input
          name="_dhs"
          type="hidden"
        />
        <button
          type="submit"
        >
          Search
        </button>
      </div>
      <div
        className="MoreOptions"
      >
        <a
          href="https://discovery.nationalarchives.gov.uk/advanced-search"
        >
          More options
        </a>
         
        or 
        <a
          href="https://discovery.nationalarchives.gov.uk/browse"
        >
          Browse
        </a>
      </div>
    </div>
  </form>
</div>
`;
